lib/src/typescript/browser.ts

```diff
@@ -72,20 +72,27 @@ export const kdf = (secret: Buffer, outputLength: number): Promise<Buffer> => {
     return willBeResult;
 }
 
-// Get the AES-128-CTR browser implementation
-const getAes = (op: typeof subtle.encrypt | typeof subtle.decrypt) => (
-    counter: Buffer,
-    key: Int8Array | Int16Array | Int32Array | Uint8Array | Uint16Array | Uint32Array | Uint8ClampedArray | Float32Array | Float64Array | DataView | ArrayBuffer,
-    data: Int8Array | Int16Array | Int32Array | Uint8Array | Uint16Array | Uint32Array | Uint8ClampedArray | Float32Array | Float64Array | DataView | ArrayBuffer
-) =>
-    subtle
-        .importKey("raw", key, "AES-CTR", false, [op.name])
-        .then(cryptoKey =>
-            op({ name: "AES-CTR", counter: counter, length: 128 }, cryptoKey, data)
```

Comment:
Instead of making two distinct full function `aesCtrEncrypt` and `aesCtrDecrypt` with lots of repetition in the code, I would have kept the `getAes` function but would have replaced this line by something like:
```javascript
if (op.name == 'encrypt')
    return subtle.encrypt({ name: "AES-CTR", counter: counter, length: 128 }, cryptoKey, data)
else
    return subtle.decrypt({ name: "AES-CTR", counter: counter, length: 128 }, cryptoKey, data)
```
But maybe I'm too simple.

Reply:
To me it's more strict and clear to have two functions, it avoids having complex types in parameters. As for the code size, it will be zipped anyway.